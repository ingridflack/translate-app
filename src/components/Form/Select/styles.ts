import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;

  select {
    position: relative;
    display: inline-block;
    margin-bottom: 15px;
    width: 100%;

    display: inline-block;
    width: 100%;
    cursor: pointer;
    padding: 11px 13px;
    outline: 0;
    border: none;
    border-bottom: 1px solid #bdbdbd;

    &:hover {
      background: #f0f1f1;
    }

    &:disabled {
      opacity: 0.5;
      pointer-events: none;
    }
  }
`;

export const SelectBox = styled.select``;
