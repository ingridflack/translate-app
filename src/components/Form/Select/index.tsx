import { HTMLAttributes } from "react";

import { Container } from "./styles";

interface SelectProps extends HTMLAttributes<HTMLSelectElement> {
  name: string;
  options: SelectOption[];
}

interface SelectOption {
  value: string;
  text: string;
}

const Select = ({ name, options, ...props }: SelectProps) => {
  return (
    <Container>
      <select name={name} id={name} {...props}>
        {options.map((option) => (
          <option value={option.value} key={option.value}>
            {option.text}
          </option>
        ))}
      </select>
    </Container>
  );
};

export default Select;
