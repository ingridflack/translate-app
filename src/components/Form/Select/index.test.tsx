import { fireEvent, render, screen } from "@testing-library/react";

import Select from ".";

const mocked_options = [
  {
    value: "value1",
    text: "text1",
  },
  {
    value: "value2",
    text: "text2",
  },
];

describe("<Select/>", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Select name="TesteName" options={mocked_options} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should render all options correctly", () => {
    render(<Select name="TesteName" options={mocked_options} />);

    const element = screen.getAllByRole("option");

    expect(element).toHaveLength(2);
  });

  it("should call the onChangeFn and change the selected option", () => {
    const onChangeFn = jest.fn();

    render(
      <Select name="TesteName" options={mocked_options} onChange={onChangeFn} />
    );

    const select = screen.getByRole("combobox");

    const [firstOption, secondOption] =
      screen.getAllByRole<HTMLOptionElement>("option");

    expect(firstOption.selected).toBeTruthy();
    expect(secondOption.selected).toBeFalsy();

    fireEvent.change(select, { target: { value: "value2" } });

    expect(firstOption.selected).toBeFalsy();
    expect(secondOption.selected).toBeTruthy();

    expect(onChangeFn).toHaveBeenCalled();
  });
});
