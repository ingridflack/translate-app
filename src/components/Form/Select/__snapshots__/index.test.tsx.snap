// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> should render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL bDCRaq"
  >
    <select
      id="TesteName"
      name="TesteName"
      role="combobox"
    >
      <option
        value="value1"
      >
        text1
      </option>
      <option
        value="value2"
      >
        text2
      </option>
    </select>
  </div>
</div>
`;
